### PR TITLE
Diberry/1031 env update

### PR DESCRIPTION
**Pull request by diberry**

### CODE REVIEW

File: components/Question.tsx [hunk 1]
```diff
@@ -69,10 +74,18 @@ export const Question: NextPage<{ count: number }> = ({ count }) => {
 
   useEffect(() => {
     if (data) {
+      console.log(`data as question ${JSON.stringify(data)}`)
```
Comment (aaronpowell) [inline components/Question.tsx]:
```suggestion
```

File: components/Question.tsx [hunk 2]
```diff
@@ -83,24 +96,33 @@ export const Question: NextPage<{ count: number }> = ({ count }) => {
...
       />
       <p className={styles.description}>
         {!correctAnswer && (
           <button
             disabled={!answer || validateAnswerLoading}
-            onClick={() =>
+            onClick={() => {
+
+              console.log(`submit`)
+
```
Comment (aaronpowell) [inline components/Question.tsx]:
```suggestion
```

File: components/Question.tsx [hunk 3]
```diff
@@ -119,12 +141,27 @@ export const Question: NextPage<{ count: number }> = ({ count }) => {
...
       </p>
+      </>
+      }
+      { questionsAskedCount >= gameQuestionCount && 
+        <div className={styles.endgamecard}>
+          <div className={styles.description}>{correctQuestionCount} correct answers in {gameQuestionCount} questions</div>
+          <button  onClick={async () => {
+
+              setQuestionsAskedCount(0)
+              setCorrectQuestionCount(0)
```
Comment (aaronpowell) [inline components/Question.tsx]:
```suggestion
          <button onClick={async () => {
              setQuestionsAskedCount(0);
              setCorrectQuestionCount(0);
```

File: package.json [hunk 4]
```diff
@@ -3,11 +3,12 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "dev": "next dev",
+    "dev": "cross-env NODE_OPTIONS='--inspect' next dev",
```
Comment (aaronpowell) [inline package.json]: we shouldn't need this as the vscode terminal will auto-attach the debugger for any node.js process

File: pages/api/datasources/TranslatorDataSource.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        textToTranslate
+      );
+
+      // return original if wasn't translated
+      if(!translationResults || translationResults.length===0) return question;
+
+      console.log(`translation results = ${JSON.stringify(translationResults)}`);
+
+      // Get Answers
+      // @ts-ignore
```
Comment (aaronpowell) [inline pages/api/datasources/TranslatorDataSource.ts]: What's the `@ts-ignore` ignoring and can we work the types better?

File: pages/api/datasources/TranslatorDataSource.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        answers
+      );
+
+      // return original if wasn't translated
+      if(!translationResults || translationResults.length===0) return answers;
+
+      console.log(`translation results = ${JSON.stringify(translationResults)}`);
+
+      // Get translated values out of returned object
+      // @ts-ignore
```
Comment (aaronpowell) [inline pages/api/datasources/TranslatorDataSource.ts]: What's the `@ts-ignore` ignoring and can we work the types better?

File: pages/api/datasources/TranslatorDataSource.ts [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+  
+      const translationResults = await getTranslation(
+        this.#client,
+        toLanguage,
+        fromLanguage,
+        [correctAnswer]
+      );
+  
+      console.log(`translation results = ${JSON.stringify(translationResults)}`);
+
```
Comment (aaronpowell) [inline pages/api/datasources/TranslatorDataSource.ts]:
```suggestion
```

File: pages/api/datasources/TranslatorDataSource.ts [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        [correctAnswer]
+      );
+  
+      console.log(`translation results = ${JSON.stringify(translationResults)}`);
+  
+      // return original if wasn't translated
+        if(!translationResults || translationResults.length===0) return correctAnswer;
+
+      // Get translated values out of returned object
+      // @ts-ignore
```
Comment (aaronpowell) [inline pages/api/datasources/TranslatorDataSource.ts]: What's the `@ts-ignore` ignoring and can we work the types better?

File: pages/api/datasources/TranslatorDataSource.ts [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+      const translatedCorrectAnswer: string = translationResults[0].translations[0].text;
+      console.log(`translatedCorrectAnswer (${toLanguage}) ${translatedCorrectAnswer}`);
```
Comment (aaronpowell) [inline pages/api/datasources/TranslatorDataSource.ts]:
```suggestion
```

File: pages/api/plugins/debugPlugin.ts [hunk 10]
```diff
@@ -0,0 +1,21 @@
+export const DebugPlugin = {
```
Comment (aaronpowell) [inline pages/api/plugins/debugPlugin.ts:1]: I'm assuming that we don't need this long term

File: pages/api/resolvers/resolvers.ts [hunk 11]
```diff
@@ -31,17 +37,25 @@ export const resolvers = {
       return question;
     },
   },
-  Question: {
+  Question: { //overwrite field resolver
+    correctAnswer(question: any){
```
Comment (aaronpowell) [inline pages/api/resolvers/resolvers.ts]:
```suggestion
    correctAnswer(question: QuestionDbModel){
```

We can add the typing now can't we?

Same goes for line 44

File: pages/index.tsx [hunk 12]
```diff
@@ -1,27 +1,40 @@
...
 
+  const container = await getExistingContainer();
+
   const { resources: items } = await container.items
     .query(querySpec)
     .fetchAll();
 
+  console.log(`getServerSideProps ${JSON.stringify(items)}`);
+
+
```
Comment (aaronpowell) [inline pages/index.tsx]:
```suggestion
```

File: pages/index.tsx [hunk 13]
```diff
@@ -1,27 +1,40 @@
...
+      count: items[0].count, // Should equal CONSTANTS.MAX_ITEMS_IN_DATABASE (100)
     },
   };
 }
 
-const Home: NextPage<{ count: number }> = (props) => {
+const Home: NextPage<{ count: number, buildOn: string }> = (props) => {
+
+  console.log(`Home count=${props.count}`);
+
```
Comment (aaronpowell) [inline pages/index.tsx]:
```suggestion
```

File: schema/question.query.graphql [hunk 14]
```diff
@@ -0,0 +1,15 @@
+query getQuestion(
```
Comment (aaronpowell) [inline schema/question.query.graphql]: Unless there's a change to bring in something that'll import the query where it's used, we can delete it.

File: upload/azure/azureCosmosdb.js [hunk 15]
```diff
@@ -0,0 +1,81 @@
+"use strict";
```
Comment (aaronpowell) [inline upload/azure/azureCosmosdb.js]: Accidental include?

File: upload/azure/secretsConfig.js [hunk 16]
```diff
@@ -0,0 +1,18 @@
+"use strict";
```
Comment (aaronpowell) [inline upload/azure/secretsConfig.js]: Accidental include?

File: upload/azure/uploadData.js [hunk 17]
```diff
@@ -0,0 +1,90 @@
+"use strict";
```
Comment (aaronpowell) [inline upload/azure/uploadData.js]: Accidental include?

File: upload/shared/constants.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
+"use strict";
```
Comment (aaronpowell) [inline upload/shared/constants.js]: Accidental include?

**Review by aaronpowell — COMMENTED** [review]
Some tidy up from the debugging code.

Also, can you run `npm run lint` as there's some formatting clean ups needed.

File: upload/azure/azureCosmosdb.js (hunk 15 above)
Comment (diberry) [reply upload/azure/azureCosmosdb.js]: didn't mean to include a generated dir

File: upload/azure/secretsConfig.js (hunk 16 above)
Comment (diberry) [reply upload/azure/secretsConfig.js]: didn't mean to include a generated dir

File: upload/azure/azureCosmosdb.js (hunk 15 above)
Comment (diberry) [reply upload/azure/azureCosmosdb.js]: didn't mean to include a generated dir

File: upload/azure/uploadData.js (hunk 17 above)
Comment (diberry) [reply upload/azure/uploadData.js]: didn't mean to include a generated dir

File: upload/shared/constants.js (hunk 18 above)
Comment (diberry) [reply upload/shared/constants.js]: didn't mean to include a generated dir

File: pages/api/plugins/debugPlugin.ts (hunk 10 above)
Comment (diberry) [reply pages/api/plugins/debugPlugin.ts:1]: In terms of teaching a customer how to developer with GraphQL, I think its an interesting tool.

**Comment by diberry** [comment]
@aaronpowell I think I've resolved or answered all your comments. In the next sprint I'll move the source into azure-samples.

**Comment by aaronpowell** [comment]
There's a merge conflict on the `main` branch, I think the removal of the bootstrap file (which was previously refactored) is causing problems

File: README.md [hunk 19]
```diff
@@ -2,27 +2,104 @@
...
+A sample dataset, obtained from [OpenTrivia DB](https://opentdb.com/), exists in the `trivia.json` file.
+
+## Next.js and Static Web App Architecture
+
+This app demonstrates a Next.js app with server-side rendering (SSR), integrated with GraphQL and Azure services (Translator and Cosmos DB). The architecture is built to show specific integrations and shouldn't be considered best practices.
+
+Learn more:
+
+- [Next.js]()
+- [Static Web Apps]()
```
Comment (aaronpowell) [inline README.md:22]: Need some links here

File: README.md [hunk 20]
```diff
@@ -2,27 +2,104 @@
...
+- Translator Text
+- TypeScript
+
+## Prerequisites
+
+The following prerequisites should be created or imstalled.
+
+- Docker for the devcontainer
+- Without Docker
+  Node.js 14 (you can use `nvm use` against the .nvmrc in the repo)
```
Comment (aaronpowell) [inline README.md:33]: I think it's Node 16 as a minimum for Next.js

**Review by aaronpowell — COMMENTED** [review]
Is the goal of this to merge into this repo or are you going to keep it separate?

I'm just checking as there is a reference to "the original source" in the readme that points back here.

**Pull request closed without merging**